### ISSUE TITLE
Create Prototype Fedora6/Rails App

### ISSUE THREAD

**Issue by bwatson78** [emory-libraries/OpenEmory#62]
Create a Rails application that integrates [Samvera's Valkyrie gem ](https://github.com/samvera/valkyrie) as a prototype site to discover how Fedora6 objects interact with the out-of-the-box adapter.

**Comment by bwatson78** [comment]
Application's code can be found here: https://github.com/emory-libraries/fedora6-prototype. Work to successfully incorporate CircleCI is ongoing.

**Comment by bwatson78** [comment]
CircleCI is working.

**Comment by eporter23** [comment]
Acceptance criteria for a first pass:

- [ ] Browse based interface that provides access to all objects migrated and ability to see more detail about each object migrated
- [ ] Basic metadata about each object shows Fedora 3 PID, Title, and new Fedora 6 ID (if different than the Fedora 3 PID)
- [ ] Ability to see and download each file migrated for the object (e.g. its data stream files)

**Comment by bwatson78** [comment]
http://10.65.19.21:8080/fcrepo-webapp-6.4.0/rest/ is live but no Fedora3-migrated objects are in this container. This issue is blocked until migration takes place.

**Comment by tmiles2** [comment]
@bwatson78 Can you summarize what you've done for the app?

**Comment by bwatson78** [comment]
Status of Prototype App As of 4/18/2023:

A Rails 7.0 application built off of Ruby 3.1.2 is working on my local machine. I have added rSpec (test suite) and rubocop (ruby's linter) as quality assurance, and that basic structure's code can be viewed at https://github.com/emory-libraries/fedora6-prototype in the `main` branch. My local machine is further along in the process, but I will push up the changes I have on my computer to Github after I compose this comment. I have also stubbed out the Capistrano deploy process in this repository, but have not actually deployed to the webapp's server because I'm currently working off of OCFL created objects that may not resemble Fedora 3 migrated objects when they are stored in Fedora 6.

I have consulted with Trey Pendragon about some anomalies I found trying to access Fedora objects via Samvera's Valkyrie adapters. He has assisted in altering some of the site's configurations so that some of the missing expected interactivity now works better, but mentioned that the current objects aren't created in a fashion that Valkyrie expects. Specifically, Valkyrie assumes that the Fedora objects it interacts with are created with UUIDs in a pair-tree structure. That is currently not the case.

Otherwise, I have been able to pull metadata and provide links to associated documents contained within objects that are housed at the base level of my Fedora 6's test instance. However, for objects that are located within a container located off of the base level, Valkyrie is not creating ruby objects for those objects, making them impossible programmatically to interact with. Collin, Emily, and I will be meeting with Trey Friday at 1230p to discuss the amount of work necessary to modify Valkyrie to parse the objects we can't interact with.

If I deployed today, there would be an index page listing two record objects that would load show pages complete with metadata and links to documents successfully. But there would be a Collection object there that, when clicked, would display a show page with two links to show pages that err when attempting to load. As soon as our test migration takes place, I would most likely have to refactor all of the code that serves these pages to properly interact with migrated objects.

**Comment by bwatson78** [comment]
In order for the app to deploy, we had to:

- install `libyaml-devel` to the server.
- copy the ssh keys (public and private) to my machine due to capistrano ssh forwarding issues.

The site isn't live at the moment due to a ruby version mismatch (2.8 vs 3.1.2)
